### PR TITLE
tools: update docs for Lightning

### PR DESCRIPTION
**Pull request by kennytm**
1. Updated the config to include the new `io-concurrency` setting, and delete the `region-min-size` setting.
2. Included several new metrics introduced recently.
3. **Removed description about mixed deployment** (putting Lightning and Importer together). This deployment mode causes too many problem in the past and is no longer officially supported.
4. Clarified some SSD and network requirement with explicit numbers.
5. **Clarified the space requirement of the TiKV cluster** (6 × SQL size).

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @lilin90 @GregoryIan

File: tools/lightning/deployment.md [hunk 1]
```diff
@@ -21,18 +21,15 @@ Before starting TiDB-Lightning, note that:
...
-
-### Hardware requirements of separate deployment
+`tidb-lightning` and `tikv-importer` are both resource-intensive programs. It is recommended to deploy them into two separate machines.
 
 To achieve the best performance, it is recommended to use the following hardware configuration:
 
 - `tidb-lightning`:
 
     - 32+ logical cores CPU
-    - 16 GB+ memory
```
Comment (kennytm) [inline tools/lightning/deployment.md]: I've also remove the memory requirement as the RAM usage of the recent version of Lightning is often below 256 MB.

**Comment by GregoryIan** [comment]
LGTM

**Comment by kennytm** [comment]
@lilin90 Changes applied

**Pull request merged**